### PR TITLE
Fix multiple cases of unobserved async errors being thrown in fire-and-forget usages

### PR DESCRIPTION
**Pull request by peppy**
This should (hopefully) cover all the cases where a SignalR exception could arrive as an unobserved exception due to there being no `await` or `ContinueWith` on an async call.

- [x] Depends on #11265 for ease of merging.

### CODE REVIEW

File: osu.Game/Screens/Multi/RealtimeMultiplayer/RealtimeMatchSongSelect.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ protected override bool OnStart()
 
                 client.ChangeSettings(item: item).ContinueWith(t =>
                 {
-                    return Schedule(() =>
+                    Schedule(() =>
```
Comment (bdach) [inline osu.Game/Screens/Multi/RealtimeMultiplayer/RealtimeMatchSongSelect.cs:61]: Is there any semantic difference with this change, other than that the underlying task changes type (from `Task<ScheduledDelegate>` to `Task`)? Not really bothered either way, just curious.

**Comment by bdach** [comment]
I've also resolved conflicts here. Was a bit less straightforward but still not too bad, but double check would be appreciated.

File: osu.Game/Screens/Multi/RealtimeMultiplayer/RealtimeMatchSongSelect.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Screens/Multi/RealtimeMultiplayer/RealtimeMatchSongSelect.cs:61]: just couldn't see a reason fort he returning of the schedule.

**Review by bdach — DISMISSED** [review]

**Review by bdach — APPROVED** [review]

**Comment by peppy** [comment]
Post-merge looks fine.

**Pull request merged**